addons/survey/views/survey_templates_statistics.xml

```diff
@@ -3,357 +3,344 @@
+                <table class="table table-hover table-sm text-right">
                     <thead>
                         <tr>
-                            <th>#</th>
-                            <th>User Responses</th>
+                            <th></th>
+                            <th class="text-right" t-foreach="table_data[0]['columns']" t-as="column_data">
+                                <span t-esc="column_data['suggested_answer'].value"></span>
+                            </th>
                         </tr>
                     </thead>
                     <tbody>
-                        <tr class="d-none" t-foreach="number_result['input_lines']" t-as="user_input">
-                            <td><a t-att-href="'/survey/print/%s?answer_token=%s' % (user_input.survey_id.access_token, user_input.user_input_id.access_token)"><t t-esc="user_input_index + 1"></t></a></td>
-                            <td><span t-field="user_input.value_numerical_box"></span><br/></td>
+                        <tr t-foreach="table_data" t-as="choice_data">
+                            <td>
+                                <span t-esc="choice_data['row'].value"></span>
+                            </td>
+                            <td class="survey_answer" t-foreach="choice_data['columns']" t-as="column_data">
+                                <span t-esc="round(column_data['count'] * 100.0/ (len(question_data['answer_line_done_ids']) or 1), 2)"></span> %
+                                <span class="badge badge-primary" t-esc="column_data['count']"></span>
+                                <i class="fa fa-filter text-primary survey_filter"
+                                    t-att-data-question_id="question.id"
+                                    t-att-data-row_id="choice_data['row'].id"
+                                    t-att-data-answer_id="column_data['suggested_answer'].id" role="img" aria-label="Survey filter" title="Survey filter"></i>
+                            </td>
                         </tr>
                     </tbody>
                 </table>
-               <t t-call="survey.question_pagination"/>
             </div>
+        <div t-if="comment_lines" t-call="survey.question_result_comments_new" /><!-- handle comments to matrix -->
         </div>
     </template>
 
-    <template id="question_pagination" name="Survey: statistics pages pagination">
-        <t t-set="record_limit" t-value="10"/><!-- Change This record_limit to change number of record  per page-->
-        <ul t-att-id="'pagination_%d' % question.id" class="pagination" t-att-data-question_id="question.id" t-att-data-record_limit="record_limit">
-            <t t-if="len(text_result) > record_limit">
-                <li t-foreach="page_range(len(text_result), record_limit)" t-as="num" class="page-item">
+    <template id="question_result_comments_new" name="Question: comments">
```

Comment:
"_new" ?

Maybe you used that name to have to two templates co-existing while testing?